src/include/api.h

```diff
@@ -198,7 +198,8 @@
 
 #define CURSOR_API_CALL(cur, s, n, bt)                                                     \
     (s) = (WT_SESSION_IMPL *)(cur)->session;                                               \
-    SESSION_API_PREPARE_CHECK(s, WT_CURSOR, n);                                            \
+    if (!F_ISSET(s, WT_SESSION_HS_CURSOR))                                                 \
```

Comment:
@sulabhM The issue is that the `set_key` call is failing when we're in a prepared transaction (which causes the key to be left as `NULL`). You can check `stderr.txt` after running the test to better understand what's happening here.

I don't know if there is a better solution but I've just pushed this change to communicate what the issue is.

Reply:
I tried several other options, none worked:
1. tried to check id the given "bt" is a HS btree
2. extracted btree out of cursor and session and check if it is HS btree

I think for now we should go with the solution @tetsuo-cpp proposed.

Reply:
I can't figure out from the report here what test is failing, but see `__txn_search_prepared_op()` for a possibly better solution:
```
    switch (op->type) {
    case WT_TXN_OP_BASIC_COL:
    case WT_TXN_OP_INMEM_COL:
        ((WT_CURSOR_BTREE *)cursor)->iface.recno = op->u.op_col.recno;
        break;
    case WT_TXN_OP_BASIC_ROW:
    case WT_TXN_OP_INMEM_ROW:
        /*
         * Transaction prepare is cleared temporarily as cursor functions are not allowed for
         * prepared transactions.
         */
        F_CLR(txn, WT_TXN_PREPARE);
        __wt_cursor_set_raw_key(cursor, &op->u.op_row.key);
        F_SET(txn, WT_TXN_PREPARE);
        break;
    case WT_TXN_OP_NONE:
    case WT_TXN_OP_REF_DELETE:  
    case WT_TXN_OP_TRUNCATE_COL:
    case WT_TXN_OP_TRUNCATE_ROW:
        WT_RET_ASSERT(session, false, WT_PANIC, "invalid prepared operation update type");
        break;
    }   
```

Reply:
@keithbostic It was the `test_hs06:test_prepare_reads`. The issue is caused by the fact that we don't allow any cursor operations while a transaction is in the prepared state. The only things you can do are commit or rollback.

Since we're not wrapping our history store inserts into a transaction, we shouldn't be subject to this restriction.

Reply:
I'm just nothing that if you clear/set the `WT_TXN_PREPARE` flag around the cursor operations, you don't need to modify `api.h`. Let me know if you have questions.

It seems like we shouldn't be using a prepared `WT_TXN` structure, if our history store inserts are done outside of a transaction, but I don't know how that design is supposed to work.